yjit/src/backend/x86_64/mod.rs

```diff
@@ -105,6 +105,33 @@ impl Assembler
     // These are the callee-saved registers in the x86-64 SysV ABI
     // RBX, RSP, RBP, and R12–R15
 
+    fn x86_merge(mut self) -> Assembler {
```

Comment:
I would also like to have a comment explaining what this pass is for. I know it seems intuitive right now, but have to think from the perspective of people seeing this codebase from the first time.

Reply:
addressed in https://github.com/ruby/ruby/pull/7453/commits/a6949591c1fe524a1b0331958cd5a6bcc8187517